### ISSUE TITLE
Different behavior when calling Fit() on a transformer chain and on an IDataLoaderEstimator

### ISSUE THREAD

**Issue by yaeldekel** [dotnet/machinelearning#2866]
The following code runs without errors:

```
            var loader = ml.Data.CreateTextLoader<InputData>(hasHeader: true, dataSample: file);
            var data = loader.Load(file);

            // Pipeline.
            var pipeline = ml.BinaryClassification.Trainers.GeneralizedAdditiveModels();

            // Train.
            var model = pipeline.Fit(data);
```
However, the following code fails with schema mismatch exception:
```
            var loader = ml.Data.CreateTextLoader<InputData>(hasHeader: true, dataSample: file);

            // Define the same pipeline starting with the loader.
            var pipeline = loader.Append(ml.BinaryClassification.Trainers.GeneralizedAdditiveModels());

           // Train
            var model = pipeline.Fit(file);
```

This may also be related to issue #1969 .

**Comment by yaeldekel** [comment]
By the way, this is the definition of `InputData`:
```
        private class InputData
        {
            [LoadColumn(0)]
            public float Label { get; set; }
            [LoadColumn(9, 14)]
            [VectorType(6)]
            public float[] Features { get; set; }
        }
```

**Comment by Ivanidzo4ka** [comment]
Does it complain on expected bool got float?
If it does I have PR which forces check input schema for first case. https://github.com/dotnet/machinelearning/pull/2804

**Comment by yaeldekel** [comment]
I think your PR may fix this issue as well as #1969, will try it again once it is merged.
